src/cmd/hotspots/assert.rs

```diff
@@ -78,7 +83,7 @@ impl Cmd {
         let wallet_key = keypair.public_key();
         let hotspot = helium_api::hotspots::get(&client, &self.gateway.to_string()).await?;
         // Get the next likely gateway nonce for the new transaction
-        let nonce = hotspot.speculative_nonce + 1;
+        let nonce = self.nonce.unwrap_or(hotspot.speculative_nonce + 1);
```

Comment:
out of interest, what caused this to be needed?

Reply:
I had a situation where the testnet etl was wedged over the weekend, as a result, the API was returning the wrong nonce I needed for a location assert, so I had to use this to overcome it.

Reply:
> Also, I think it's important to be able to set the nonce for every transaction, as that is what will be needed for air-gapped wallet operation, which I believe we generally support, correct?

Easier said then done but air-gapped support is what I thought you were going for which this PR does not quite achieve. 

> 
> Irrespective, of course, that the current code does have an explicit query to the API, at the moment.

We've only really added air-gap support to specific wallet commands as the community needs them since it can get quite involved. This was not one of them